lib/streamlit/type_util.py

```diff
@@ -445,11 +453,16 @@ def convert_anything_to_df(df: Any) -> DataFrame:
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
-    if is_type(df, _SNOWPARK_DF_TYPE_STR) and not isinstance(df, list):
-        df = pd.DataFrame(df.take(MAX_UNEVALUATED_DF_ROWS))
-        if df.shape[0] == MAX_UNEVALUATED_DF_ROWS:
+    if (
+        is_type(df, _SNOWPARK_DF_TYPE_STR)
+        and not isinstance(df, list)
+        or is_type(df, _SNOWPARK_TABLE_TYPE_STR)
+    ):
+        df = pd.DataFrame(df.take(max_unevaluated_rows))
+        if df.shape[0] == max_unevaluated_rows:
             st.caption(
-                f"⚠️ Showing only 10k rows. Call `collect()` on the dataframe to show more."
+                f"⚠️ Showing only {'10k' if max_unevaluated_rows == MAX_UNEVALUATED_DF_ROWS else str(max_unevaluated_rows)} rows. "
```

Comment:
nit: at this point, it might be worth adding a function to `string_util` that automatically does this, e.g.:

```python
def simplify_number(num: int) -> str:
    num_converted = float("{:.2g}".format(num))
    magnitude = 0
    while abs(num_converted) >= 1000:
        magnitude += 1
        num_converted /= 1000.0
    return "{}{}".format(
        "{:f}".format(num_converted).rstrip("0").rstrip("."),
        ["", "k", "m", "b", "t"][magnitude],
    )
```

Reply:
Your func is created in `string_util` and units tests are added 👍 Thank you 🙇